plugins/processors/k8sdecorator/stores/utils.go

```diff
@@ -30,13 +31,23 @@ func createContainerKeyFromMetric(tags map[string]string) string {
 	return k8sutil.CreateContainerKey(namespace, podName, containerName)
 }
 
-const (
-	// kubeAllowedStringAlphaNums holds the characters allowed in replicaset names from as parent deployment
+var (
+	// deploymentUnexpectedRegEx holds the characters allowed in replicaset names from as parent deployment
 	// https://github.com/kubernetes/apimachinery/blob/master/pkg/util/rand/rand.go#L83
-	kubeAllowedStringAlphaNums = "bcdfghjklmnpqrstvwxz2456789"
-	cronJobAllowedString       = "0123456789"
+	deploymentUnexpectedRegEx = regexp.MustCompile(`[^b-hj-np-tv-xz-z2-24-9]+`)
+	// cronJobUnexpectedRegex ensures the characters in cron job name are only numbers.
+	cronJobUnexpectedRegex = regexp.MustCompile(`^[^0-9]+$`)
 )
 
+//Defining Embedding type to access to the RegExp methods (https://stackoverflow.com/a/43507669) since we cannot define new methods to non-local type
+type Regexp struct {
+	*regexp.Regexp
+}
+
+func (unexpectedRegEx *Regexp) containUnexpectedRuneSet(name string) bool {
```

Comment:
Rather than defining a function of a struct like this, we could have just changed the `containUnexpectedRuneSet()` function to accept a regex as a parameter.
```suggestion
func containsUnexpectedRuneSet(name string, r *regexp.Regexp) bool {
```

Then further down, we can just use the regular expressions compiled as vars for the check:
```go
if containsUnexpectedRuneSet(suffix, cronJobUnexpectedRegex) {
```

Reply:
Reasonable. Fixed it in the newest commit though. Thought it would be better to defining new structure type and thought of get ridding of the function at a later point. Seeing this make more sense to me though